### ISSUE TITLE
Allow lan with exit node

### ISSUE THREAD

**Issue by engineeringsys** [tailscale/tailscale#3268]
### What is the issue?

I've got a properly configured exit node which by default when I bring tailscale0 up on a node and pass the arg flag of --exit-node=exit-node-ip will of course route 0.0.0.0/::0 through the exit as it should.

I was looking into the proper use `--exit-node-allow-lan-access, --exit-node-allow-lan-access=true`

When trying this on the client node I would execute:
sudo tailscale up  --authkey <my auth key> --exit-node=exit-node-ip --exit-node-allow-lan-access=true --reset --force-reauth

The assumed behavior based on what i know is that it will exit all traffic except for lan traffic but the actual behavior prevents me from ssh'ng to the machine via the lan which I do have a direct route to on my VLAN.

I'm hoping I'm misinterpreting the way this works but was also hoping someone could shed some light on this.

I'm on all linux clients at the moment, I'm thinking this lan allow flag may have to go into the exit node, but I'm not sure.

### OS

Linux

### OS version

Arch Linux latest

### Tailscale version

1.16.1

**Comment by DentonGentry** [comment]
Enabling an exit node without using --allow-lan-access blocks all egress and ingress traffic from the LAN.
Certainly, enabling `--exit-node=exit-node-ip` without setting `--exit-node-allow-lan-access` would be expected to block SSH access.

One thing I note: `sudo tailscale up --authkey=tskey-... --force-reauth` will have it authenticate and get a new node key, which means its Tailscale IP address will change. You're SSHing to the LAN IP address, not the Tailscale IP address?

**Comment by engineeringsys** [comment]
Good call on the second part, I should negate the `--force-reauth`

So if I'm understanding this correctly (forgive me, it's been a long few days)

Exit node setup:

`sudo tailscale up --advertise-exit-node --exit-node-allow-lan-access --authkey <authkey>`

Client node setup:
`sudo tailscale up --exit-node=<exitnodetailscaleip> --authkey <authkey>`

I believe I tried the above and cannot route ipv4 traffic anywhere.  DNS is a whole other story but I'm equipped to debug that.  I think I should start in a clean vm and vps since I may have persistent configuration on my filesystem.

**Comment by engineeringsys** [comment]
I tried enabling the exit node by doing the `--advertise-exit-node` and passing `--exit-node-allow-lan-access`  but it notes `--exit-node-allow-lan-access must be ran with --exit-node` so I'm assuming this is a client side arg flag to be passed?

I tried it on a new local (lan) kvm, using the exit node tailscale ip after activate the exit node in the ui and it's not-reachable via lan

**Comment by maisem** [comment]
>I tried enabling the exit node by doing the --advertise-exit-node and passing --exit-node-allow-lan-access but it notes --exit-node-allow-lan-access must be ran with --exit-node so I'm assuming this is a client side arg flag to be passed?

Yes it's a client side flag

@engineeringsys the machine that you're trying to ssh into is on a different subnet than the machine that you are trying to connect from, correct? Or are you trying to reach a machine which is on the lan of the exit node?

For the former, as tailscale has no way of knowing which other private networks are accessible from your lan, it doesn't handle those routes. e.g. say my machine is on `192.168.0.0/24` and the one I am trying to access is on `192.168.10.0/24`.
With `--exit-node=...`, tailscale will route all traffic including `192.168.0.0/24` via the exit node
With `--exit-node=... --exit-node-allow-lan-access`, tailscale will route all traffic **except** `192.168.0.0/24` (and any other configured subnets on the client) via the exit node

For the latter, have you considered using [subnet routers](https://tailscale.com/kb/1019/subnets/)?

**Comment by engineeringsys** [comment]
@maisem @DentonGentry 

🤦‍♂️ I'm so sorry.  I should have explained that I have multiple-vlans and where I was testing from was on another vlan.

I plugged my laptop into the vlan I wanted to mesh from, set it to exit out into a vps (exit node) with the allow lan access configuration, could send icmp/tcp traffic via lan to that subnet as well as fully exit as expected.

I'm not sure where I went wrong in my debug phase.  Perhaps I had a problem interpreting docs.  I'm going to write something up like a set of notes with caveats for myself.  If there's an opportunity for a doc update will do a PR as well.

Thank you both for being patient.
